packages/app/obojobo-document-engine/src/scripts/viewer/components/viewer-app.js

```diff
@@ -341,22 +345,44 @@ export default class ViewerApp extends React.Component {
 			}).then(res => {
 				this.leaveEvent = res.value
 			})
-		} else {
+
+			return
+		}
+
+		// From Hiding to Viewing
+		if (this.viewerHideDate) {
+			// leaveEvent may not exist if postEvent for 'viewer:leave' didn't complete
+			const relatedEventId = this.leaveEvent?.extensions?.internalEventId ?? 'not availible'
+
 			ViewerAPI.postEvent({
 				draftId: this.state.model.get('draftId'),
 				action: 'viewer:return',
 				eventVersion: '2.0.0',
```

Comment:
```suggestion
				eventVersion: '3.0.0',
```

My thought for this - `relatedEventId` used to expect null or some id, now it may report "not available". I'm open to discussion on that though.